akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -121,7 +121,7 @@ import scala.concurrent.{ Future, Promise }
           if (listener ne null) listener ! ResumeAccepting(1)
         }
 
-        override def onDownstreamFinish(): Unit = tryUnbind()
+        override def onDownstreamFinish(cause: Throwable): Unit = tryUnbind()
```

Comment:
Maybe log something here if cause is failure? I could imagine that you could be surprised when your server unbinds because of some downstream error.

Reply:
Wouldn't that already be logged by the failing downstream?

Reply:
Thinking about it a bit more it makes sense, I'll add debug logging for non-failure cancellation as well to make it less opaque

Reply:
Yeah it might be logged but often people seem to drop the result of `runForeach` so they might be surprised about the server unbinding.

Reply:
I think it should be debug-logged in all cases btw. Same below.